doc/sphinx/language/core/modules.rst

```diff
@@ -1010,3 +1010,66 @@ subdirectories of path). See the command :cmd:`Declare ML Module` in
 
 See :ref:`command-line-options` for a more general view over the Coq command
 line options.
+
+.. _controlling-locality-of-commands:
+
+Controlling the scope of commands with locality attributes
+----------------------------------------------------------
+
+Many commands produce an effect that will persist in a given scope,
+which can typically be the one of the section or of the module in
+which the command was called. Locality :term:`attributes <attribute>`
+can be used to change the scope of this effect. Note that some
+commands, like :cmd:`Arguments` do not have the same default locality
+when they are used inside or outside sections.
+
+.. attr:: local
+
+   In most cases, the :attr:`local` attribute limits the effect of the
+   command to the current scope (section or module).
+
+   The only exception is when applied to :cmd:`Definition`,
+   :cmd:`Theorem` or variants. For these commands, the effect of the
+   :attr:`local` attribute will be that the defined object will not be
+   available through a shorter name when importing the module in which
+   it is defined.
+
+   There is an alternative syntax for the :attr:`local` attribute
+   using the ``Local`` prefix (see :n:`@legacy_attr`).
+
+   .. note::
+
+      For some commands, this is the only supported locality within
+      sections (e.g., for :cmd:`Notation`, :cmd:`Ltac` and :ref:`Hint
+      <creating_hints>` commands).
+
+.. attr:: export
+
+   The :attr:`export` attribute makes the effect of the command
+   persist when the section is closed and when the module containing
+   the command is imported.
+
+   This attribute is supported in particular by the :ref:`Hint
+   <creating_hints>`, :cmd:`Set` and :cmd:`Unset` commands, although
+   for :ref:`Hint <creating_hints>` commands, it is not supported
+   inside sections.
+
+.. attr:: global
+
+   In most cases, the :attr:`global` attribute makes the effect of the
+   command persist even when the current section or module is closed
+   and in other files as soon as the one containing the command is
+   (possibly transitively) loaded.
+
+   For a few commands (like :cmd:`Notation` and :cmd:`Ltac`), this
+   attribute behaves as in the description of :attr:`export`.
+
+   There is an alternative syntax for the :attr:`global` attribute
+   using the ``Global`` prefix (see :n:`@legacy_attr`).
+
+   .. warning::
+
+      Using the :attr:`global` locality is strongly discouraged
+      because the transitive nature of file loading gives little
+      control to the user. Prefer the use of the :attr:`export`
+      locality attribute when it is supported.
```

Comment:
```suggestion
      We strongly discourage using :attr:`global` in commands
      because the transitive nature of file loading gives the user little
      control. We recommend using the :attr:`export`
      locality attribute where it is supported.
```